### PR TITLE
Some structural changes after updating the flowcharts

### PR DESCRIPTION
**Pull request by timovdk**
We (@Ratoone, @RobotJesse and I) made some changes after updating the flowcharts

### CODE REVIEW

**Review by RobotJente — APPROVED** [review]
Great! 🥇

**Comment by timovdk** [comment]
Locally this branch builds fine, just waiting for Travis to be fixed before I merge this

File: src/stp/Play.cpp
Comment (Yuhanun) [inline src/stp/Play.cpp]:
```suggestion
        stpInfos.clear();
```

**Review by Yuhanun — CHANGES_REQUESTED** [review]

**Review by Yuhanun — APPROVED** [review]
lgtm

**Pull request merged**
